tests/feature/ExportDumpTest.php

```diff
@@ -128,4 +163,173 @@ public function createsStreamedFileDownloadResponse()
         $this->assertInstanceOf(StreamedResponse::class, $response);
         $this->assertEquals(200, $response->getStatusCode());
     }
+
+    /**
+     * @test
+     * @dataProvider provideForHasCorrectConfiguration
+     */
+    public function hasCorrectConfiguration(array $protectorOptions, array $expected): void
+    {
+        $this->configureProtector($protectorOptions);
+
+        $connection = DB::connection($this->protector->getConnectionName());
+        $schemaState = $connection->getSchemaState();
+        $schemaStateProxy = $this->runProtectedMethod('getProxyForSchemaState', [$schemaState]);
+
+        $conditionalParameters = $schemaStateProxy->getConditionalParameters();
+
+        $this->assertEquals($expected[$connection->getDriverName()], $conditionalParameters);
+    }
+
+    public static function provideForHasCorrectConfiguration(): array
+    {
+        return [
+            'all options on' => [
+                'protectorOptions' => [
+                    self::PROTECTOR_WITH_CREATE_DB,
+                    self::PROTECTOR_WITH_DROP_DB,
+                    self::PROTECTOR_WITH_COMMENTS,
+                    self::PROTECTOR_WITH_CHARSETS,
+                    self::PROTECTOR_WITH_DATA,
+                    self::PROTECTOR_WITH_TABLESPACES,
+                ],
+                'expected' => self::getExpected([
+                    'pgsql' => [
+                        self::POSTGRES_CREATE => true,
+                        self::POSTGRES_CLEAN => true,
+                        self::POSTGRES_VERBOSE => true,
+                        self::POSTGRES_SCHEMA_ONLY => false,
+                        self::NO_TABLESPACES => false,
+                    ],
+                    'mysql' => [
+                        self::MYSQL_SET_GTID_PURGED => true,
```

Comment:
so this is always true because it cant be set in protector and we dont test mariadb?

Reply:
exactly